src/persist-client/src/fetch.rs

```diff
@@ -405,36 +425,19 @@ where
     ///
     /// !!!WARNING!!!
     ///
-    /// This semantically transfers the lease to the returned
-    /// SerdeLeasedBatchPart. If `self` has a `leased_seqno`, failing to take
-    /// the returned `SerdeLeasedBatchPart` back into a `LeasedBatchPart` will
-    /// leak `SeqNo`s and prevent persist GC.
-    pub fn into_exchangeable_part(mut self) -> SerdeLeasedBatchPart {
+    /// This method also returns the [Lease] associated with the given part, since
+    /// that can't travel across process boundaries. The caller is responsible for
+    /// ensuring that the lease is held for as long as the batch part may be in use:
+    /// dropping it too early may cause a fetch to fail.
+    pub fn into_exchangeable_part(mut self) -> (SerdeLeasedBatchPart, Lease) {
         let (proto, _metrics) = self.into_proto();
         // If `x` has a lease, we've effectively transferred it to `r`.
-        let _ = self.leased_seqno.take();
-        SerdeLeasedBatchPart {
+        let lease = std::mem::take(&mut self.lease);
```

Comment:
The disconnected `Default` impl is quite clever, but I think `Option<Lease>` is more obvious (e.g. if you Debug print something where it's been taken, you'll see a `None`